### ISSUE TITLE
List continuation indentation

### ISSUE THREAD

**Issue by serussell** [neovim/neovim#10206]
There have been discussions in markup-related projects about list continuation indentation, for example [here](https://github.com/vim-pandoc/vim-pandoc-syntax/issues/161) and [here](https://github.com/reedes/vim-pencil/issues/60).  The question is whether there's a way to indent wrapped lines when the line starts with a particular sequence of characters; this would allow syntaxes to do things like indent wrapped long lines in lists, and only lists, without affecting non-list wrapped text (as would happen when using `showbreak`.

Somebody [provided a patch](https://github.com/torbiak/vim/commit/56b35db7a3c7ec0ad2712a10c98f378c9b5583f5) to the `vim` code base that apparently does this; I don't know whether it was ever adopted, but it appears not.  It's not a long patch, but I can't tell whether it's good or not.  

Would it be possible to get something that achieves the same thing into `neovim`?

**Comment by justinmk** [comment]
```
- a long list item
  that wraps
- [ ] items with checkboxes
      wrap past the box
```

I personally wouldn't want that, but anyways this type of minor tweak to Vim editor features is better done at https://github.com/vim/vim/ , it's not a good use of our time to add random features if Vim is likely to implement it upstream.

**Comment by serussell** [comment]
@justinmk 
>  if Vim is likely to implement it upstream.

The patch I linked to was submitted to the vim project two years ago; is it really likely that it'll be implemented upstream?

**Comment by justinmk** [comment]
> The patch I linked to was submitted to the vim project two years ago; is it really likely that it'll be implemented upstream?

If it has tests and documentation, then yes. Same would be required to merge it into Nvim.

Vim is not shy about adding tons of bloated features, they recently added the ability to play a sound (yes, really).

**Comment by alerque** [comment]
@serussell Given the patch author's [comments here](https://github.com/torbiak/vim/commit/56b35db7a3c7ec0ad2712a10c98f378c9b5583f5#commitcomment-34076132) I don't think it ever was actually submitted for inclusion to the vim project, it never got past the hacky proof of concept stage.

**Comment by serussell** [comment]
@alerque Hm.  Too bad; it'd have made text markup documents much easier to visually parse.

**Comment by LeuschkeTressa** [comment]
> @alerque Hm. Too bad; it'd have made text markup documents much easier to visually parse.

Yes, really. I have been looking for this feature for several years among dozens of text editor, so far without success.

It's a pity that not even vim/neoVim is configurable enough that this can be achieved, not even with a custom plugin, as I understand it.

(Note: There's [a suggestion here](https://github.com/vim-pandoc/vim-pandoc-syntax/issues/161) using `set showbreak=\ \ ` which seems to work in some or most cases. It breaks with some plugins though and probably has other drawbacks as well.)

**Comment by rightaway** [comment]
https://github.com/vim/vim/issues/1661 was just closed. Can this issue be reopened?

**Comment by clason** [comment]
No need; this patch will then be ported in due time. If you want to expedite this, consider [making a patch PR](https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim) yourself.

**Comment by groves** [comment]
Ported in https://github.com/neovim/neovim/commit/f89a275e32110a63e8ee1fc6ca75e0cf09194185

**Comment by rightaway** [comment]
Will it be included in the release after 0.5.0?

**Comment by clason** [comment]
Probably not backported to 0.5.1, but definitely in the next full release, 0.6.0.
